ssl/s3_enc.c

```diff
@@ -299,6 +299,7 @@ int ssl3_setup_key_block(SSL *s)
     s->s3->tmp.key_block = p;
 
     ret = ssl3_generate_key_block(s, p, num);
+    if (ret==0) goto err;
```

Comment:
This change is incorrect (as are the other associated changes in this function). The value of `p` has already been assigned to `s->s3->tmp.key_block` and will therefore get freed with the SSL object is freed. Freeing it now without resetting `s->s3->tmp.key_block` will result in a double free.